forta-bots/DAI-Price-Monitor/src/agent.ts

```diff
@@ -0,0 +1,101 @@
+import BigNumber from "bignumber.js";
+import { providers } from "ethers";
+import {
+  Finding,
+  getEthersProvider,
+  HandleTransaction,
+  Initialize,
+  LogDescription,
+  TransactionEvent,
+} from "forta-agent";
+import Fetcher from "./fetcher";
+import NetworkData, { NETWORK_MAP } from "./network";
+import NetworkManager from "./network";
+import { SWAP_EVENT, TOKEN_EXCHANGE_EVENT } from "./abi";
+import {
+  createFinding,
+  calculateUniswapPrice,
+  calculateCurvePrice,
+  uniswapPairCreate2,
+  SPREAD_THRESHOLD,
+} from "./utils";
+
+const networkManager: NetworkData = new NetworkManager(NETWORK_MAP);
+
+export const initialize =
+  (provider: providers.Provider): Initialize =>
+  async () => {
+    const { chainId } = await provider.getNetwork();
+    networkManager.setNetwork(chainId);
+  };
+
+export const provideHandleTransaction = (
```

Comment:
let's split the different pools logic in different files, so you can invoke each pool handler inside the main bot logic. You can write separeted tests for each one of them, and finally write tests to detect findings from many pools on the `agent.spec.ts`